WalletWasabi.Fluent/ViewModels/AddWallet/WalletNamePageViewModel.cs

```diff
@@ -97,6 +103,11 @@ protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disp
 	{
 		base.OnNavigatedTo(isInHistory, disposables);
 
+		if (!isInHistory && !UiContext.WalletRepository.HasWallet)
```

Comment:
Is the `!isInHistory` correct here? I think negating is not required. If you want to cover the case when the user navigates back from the next screen then I think the flag `isInHistory` is true, because it is only false, when we navigate to the dialog for the first time.

Reply:
You're right. Using `!isInHistory` for some reason had the correct behavior (skip back and forth between "Add New Wallet" page and "Recovery Words" page) but the stack went crazy. Fixed.